[KRW35JP3] Дашборд для гостя

AlexeyFrunza opened the pull request:
-- Реализован весь функционал страницы DashboardDisconnectedPage

Commit:
[KRW35JP3] Дашборд для гостя
-- Добавлена страница гостя
-- Добавлен виджет AuthStart
-- Добавлен виджет DashboardInfo, DashboardVideo
-- Начата верстка виджета StatGeneral

Commit:
[KRW35JP3] Дашборд для гостя
-- Добавлена страница гостя
-- Добавлен виджет AuthStart
-- Добавлен виджет DashboardInfo, DashboardVideo
-- Начата верстка виджета StatGeneral

Commit:
[KRW35JP3] Дашборд для гостя
-- Добавлена страница гостя
-- Добавлен виджет AuthStart
-- Добавлен виджет DashboardInfo, DashboardVideo
-- Начата верстка виджета StatGeneral

Commit:
[KRW35JP3] Дашборд для гостя
-- Добавлена страница гостя
-- Добавлен виджет AuthStart
-- Добавлен виджет DashboardInfo, DashboardVideo
-- Начата верстка виджета StatGeneral

Commit:
[KRW35JP3] Дашборд для гостя
-- Добавлена страница гостя
-- Добавлен виджет AuthStart
-- Добавлен виджет DashboardInfo, DashboardVideo
-- Начата верстка виджета StatGeneral

Commit:
[KRW35JP3] Дашборд для гостя
-- Добавлена страница гостя
-- Добавлен виджет AuthStart
-- Добавлен виджет DashboardInfo, DashboardVideo
-- Начата верстка виджета StatGeneral

Commit:
[KRW35JP3] Дашборд для гостя
-- добавлена сущность stat, описаны интерфейсы и сервис
для её использования
-- виджет в стадии разработки

Commit:
[KRW35JP3] Дашборд для гостя
-- добавлена сущность stat, описаны интерфейсы и сервис
для её использования
-- виджет в стадии разработки

Commit:
[KRW35JP3] Дашборд для гостя
-- добавлена сущность stat, описаны интерфейсы и сервис
для её использования
-- виджет в стадии разработки

Commit:
[KRW35JP3] Дашборд для гостя
-- добавлена сущность stat, описаны интерфейсы и сервис
для её использования
-- виджет в стадии разработки

Commit:
Merge branch 'feature/KRW35JP3' of github.com:weekkkk/flowzy-crypto into feature/KRW35JP3

Commit:
[KRW35JP3] Дашборд для гостя
-- Доделал widget StatGeneralWidget
-- Описал структуру, бизнес логику и сверстал виджет StatRecentWidget,
но столкнулся с проблемами при получении данных

Commit:
[KRW35JP3] Дашборд для гостя
-- Доделал widget StatGeneralWidget
-- Описал структуру, бизнес логику и сверстал виджет StatRecentWidget,
но столкнулся с проблемами при получении данных

Commit:
[KRW35JP3] Дашборд для гостя
-- Доработан виджет StatRecent, полностью готов функционал
-- Адаптировано под мобильные устройства: AuthStartWidget,
DashboardPageInfo, DashboardPageVideo, StatGeneralWidget

Commit:
[KRW35JP3] Дашборд для гостя
-- Доработан виджет StatRecent, полностью готов функционал
-- Адаптировано под мобильные устройства: AuthStartWidget,
DashboardPageInfo, DashboardPageVideo, StatGeneralWidget

Commit:
[KRW35JP3] Дашборд для гостя
-- Доработан функционал StatGeneralWidget
-- Добавлен адаптив для всех элементов страницы

Commit:
[KRW35JP3] Дашборд для гостя
-- Доработан функционал StatGeneralWidget
-- Добавлен адаптив для всех элементов страницы

AlexeyFrunza commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Дашборд для гостя](https://trello.com/c/KRW35JP3/15-%D0%B4%D0%B0%D1%88%D0%B1%D0%BE%D1%80%D0%B4-%D0%B4%D0%BB%D1%8F-%D0%B3%D0%BE%D1%81%D1%82%D1%8F)

weekkkk commented on public/images/play-video.svg:
убери

weekkkk commented on src/entities/stat/api/interfaces/recent-stat.record.dto.ts:
```diff
@@ -0,0 +1,5 @@
+export interface RecentStatRecDto {
```
поправить нейминг, на схеме отобразил изменения

weekkkk commented on src/entities/stat/api/service.ts:
```diff
@@ -0,0 +1,17 @@
+import { $stat } from "./point";
+
+export class StatService {
+  static async getGeneralStat() {
+    const res = await $stat<GetGeneralStatResDto>("/get-general-stat");
+    return res;
+  }
+
+  static async getRecentStat() {
+    const res = await $stat<GetRecentStatResDto>("/get-recent-stat");
+    return res;
+  }
+}
```
поправить нейминг методов, на схеме отобразил изменения

weekkkk commented on src/entities/stat/ui/card/index.vue:
```diff
@@ -0,0 +1,82 @@
+const participants = computed(() => {
+  return {
+    value: totalParticipants.value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","),
+    profit: totalParticipants.profit.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","),
```
вынеси `.replace(/\B(?=(\d{3})+(?!\d))/g, ",")` в отдельную функцию, `shared/utils`

weekkkk replied:
а потом сделай в `entities/stat/utils` функцию для парсинга записи статистики, которая будет использовать функцию из `shared`, чтобы не дублировать логику

weekkkk commented on src/entities/stat/ui/card/index.vue:
```diff
@@ -0,0 +1,82 @@
+      <p class="text-lg bg-gradient-to-r primary-gradient bg-clip-text text-transparent max-md:text-sm">
+        + {{ participants.profit }}
+      </p>
+    </div>
```
это везде дублируется, вынеси в отдельный компонент

weekkkk replied:
корнем компонента будет `<li />` элемент

weekkkk commented on src/entities/stat/ui/card/index.vue:
```diff
@@ -0,0 +1,82 @@
+</script>
+
+<template>
+  <div class="w-115 h-121.75 bg-neutral-800 max-md:bg-transparent rounded-5xl px-12.5 pt-12.5 flex flex-col max-md:w-full max-md:px-0 max-md:h-auto max-md:pt-0">
```
сделай реализацию этого компонента через теги `<ul />` `<li />` для семантики

weekkkk commented on src/entities/stat/ui/card/index.vue:
изменился нейминг, смотри на схему

weekkkk commented on src/entities/stat/ui/table/index.vue:
```diff
@@ -0,0 +1,42 @@
+        <div v-show="!isMobile" class="text-2xl">
+          <span class="text-neutral-400 opacity-60">{{ getPassedTime(item.date) }}</span>
+        </div>
+      </div>
```
это нужно реализовать через `<ul />` `<li />` для семантики

weekkkk commented on src/entities/stat/ui/table/index.vue:
изменился нейминг, смотри схему

weekkkk replied:
этот компонент должен содержать только таблицу, заголовок и контейнер должен добавляться в виджете

weekkkk commented on src/entities/stat/api/interfaces/get-recent-stat.res.dto.ts:
```diff
@@ -0,0 +1,5 @@
+import type { RecentStatRecDto } from "./recent-stat.record.dto";
+
+export interface GetRecentStatResDto {
+  data: RecentStatRecDto[];
```
назови поле `records`

Commit:
[KRW35JP3] Дашборд для гостя
-- внёс правки по схеме и по review с gh

Commit:
[KRW35JP3] Дашборд для гостя
-- внёс правки по схеме и по review с gh

weekkkk commented on src/entities/stat/api/interfaces/general-stat-record.dto.ts:
поправь нейминг файла, будет просто `general-record.dto.ts`

weekkkk commented on src/entities/stat/api/interfaces/get-general-stat.res.dto.ts:
аналогично

weekkkk commented on src/entities/stat/api/interfaces/get-recent-stat.res.dto.ts:
аналогично

weekkkk commented on src/entities/stat/api/interfaces/recent-stat.record.dto.ts:
аналогично

weekkkk commented on src/entities/stat/api/service.ts:
```diff
@@ -0,0 +1,17 @@
+  }
+}
+
+export * from "./interfaces";
```
это зачем?

weekkkk commented on src/entities/stat/ui/general-card/interfaces/props.ts:
```diff
@@ -0,0 +1,7 @@
+import type { StatGeneralRecordDto } from "../../../api/interfaces";
+
+export interface StatCardProps {
```
теперь это не просто `StatCard...`, а `StatGeneralCard...`

weekkkk commented on src/entities/stat/ui/general-card/ui/item/interfaces/props.ts:
```diff
@@ -0,0 +1,6 @@
+export interface StatCardItemProps {
```
аналогично

weekkkk commented on src/entities/stat/ui/general-card/index.vue:
```diff
@@ -0,0 +1,50 @@
+
+<template>
+  <ul class="pb-5 max-md:pb-0">
+    <StatGeneralCardItem :record="participants" class="pt-6.25 max-md:pt-0 pb-5 max-md:pb-3">
```
`class="pt-6.25 max-md:pt-0 pb-5 max-md:pb-3"` дублируется поставь внутрь компонента, а для последнего можно использовать `last:`
https://tailwindcss.com/docs/hover-focus-and-other-states#first-last-odd-and-even

weekkkk commented on src/entities/stat/ui/recent-table/interfaces/props.ts:
```diff
@@ -0,0 +1,5 @@
+import type { StatRecentRecordDto } from "../../../api/interfaces";
+
+export interface RecentStatCardProps {
```
пооправь нейминг по аналогии с каарточкой (если что тут как минимум таблица, а не карточка)

weekkkk commented on src/entities/stat/utils/format.ts:
```diff
@@ -0,0 +1,9 @@
+import type { StatGeneralRecordDto } from "../api/interfaces";
+import { format } from "~/src/shared/utils";
```
для этой функции должен быть работать автоимпорт, саму функцию назови адекватно
если она форматирует число, то так и назови `formatNumber`

weekkkk replied:
и если она форматирует число, то должна принимать число, а не строку

weekkkk commented on src/entities/stat/utils/format.ts:
```diff
@@ -0,0 +1,9 @@
+import type { StatGeneralRecordDto } from "../api/interfaces";
+import { format } from "~/src/shared/utils";
+
+export function formatRecord(rec: StatGeneralRecordDto) {
```
эту функцию назову точнее, записи могут быть разные, в данном случае `formatStatGeneralRecord`

weekkkk commented on src/entities/stat/utils/format.ts:
```diff
@@ -0,0 +1,9 @@
+
+export function formatRecord(rec: StatGeneralRecordDto) {
+  return {
+    value: format(rec.value.toString()),
```
если функция нужна для форматирования числа, значит должна принимать число

weekkkk commented on src/shared/utils/stat-record-format.ts:
```diff
@@ -0,0 +1,3 @@
+export function format(str: string) {
```
будет принимать число, и внутри преобразовывать в строку

weekkkk commented on src/widgets/stat/ui/general/index.vue:
```diff
@@ -0,0 +1,9 @@
+<script lang="ts" async setup>
+const { data } = await useAsyncData("stat", () => StatService.getGeneral(), {});
```
`{}` убери

weekkkk commented on src/widgets/stat/ui/recent/index.vue:
```diff
@@ -0,0 +1,14 @@
+<script lang="ts" async setup>
+const { data } = await useAsyncData("stat-recent", () => StatService.getRecent(), {});
```
аналогично `{}`

weekkkk committed:
DRtji433 / Инициализация
- Изменил поинт апи
-- Временное решение, как будет время пересмотрю подход к структуре `api` дирректории

Commit:
[KRW35JP3] Дашборд для гостя
-- правки по re-review
-- изменен нейминг интерфейсов, доработана утилита

Commit:
[KRW35JP3] Дашборд для гостя
-- правки по re-review
-- изменен нейминг интерфейсов, доработана утилита

Commit:
[KRW35JP3] Дашборд для гостя
-- правки по re-review
-- изменен нейминг интерфейсов, доработана утилита

Commit:
[KRW35JP3] Дашборд для гостя
-- правки по re-review
-- изменен нейминг интерфейсов, доработана утилита

Commit:
[KRW35JP3] Дашборд для гостя
-- правки по re-review
-- изменен нейминг интерфейсов, доработана утилита

Commit:
[KRW35JP3] Дашборд для гостя
-- правки по re-review
-- изменен нейминг интерфейсов, доработана утилита

KossaDenis committed:
[Hvxh7cth] - Шапка
- Правки

weekkkk reviewed (changes requested):
не все поправил, рразрешил конфликты слияния

weekkkk committed:
Merge pull request #5 from weekkkk/feature/Hvxh7cth

[Hvxh7cth] - Шапка

weekkkk committed:
Merge branch 'feature/DRtji433' into develop

KossaDenis committed:
[Hvxh7cth] - Шапка
- Правки

weekkkk committed:
Merge pull request #6 from weekkkk/feature/Hvxh7cth

[Hvxh7cth] - Шапка

Commit:
Merge branch 'feature/KRW35JP3' of github.com:weekkkk/flowzy-crypto into feature/KRW35JP3

AlexeyFrunza commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Подключение кошелька](https://trello.com/c/zON3joCI/16-%D0%BF%D0%BE%D0%B4%D0%BA%D0%BB%D1%8E%D1%87%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BA%D0%BE%D1%88%D0%B5%D0%BB%D1%8C%D0%BA%D0%B0)
